SPECS/setup/setup.spec

```diff
@@ -0,0 +1,1301 @@
+
+# Default to no check, because it pulls in other packages at build,
+# and that will cause "circular dependency" problems for the Azure
+# Linux toolkit
+%bcond check 0
```

Comment:
Why do we need to hard-code a disabled test? We shouldn't implement a separate mechanism to controlling tests runs over what we already try to do with `with_check`. What happens when you use `with_check` instead and remove this line? The circular dependencies sound worrying in this case.

Reply:
reduce delta from upstream, and i'd like to reenable it when/if we are not so restricted by "circular dependencies"

Reply:
I missed a BR for `tcsh` - that would be an issue for toolchain builds, I agree.

I think it would still be worth to enable the tests to run during non-toolchain builds, so we keep `setup` tested. How that works, is all the BRs behind `0%{with_check}` will be ignored during the non-test build and the test build of the package will wait until its dependencies are present (BRs behind `0%{with_check}` never cause circular dependencies). The `%check` failures during the toolchain build are ignored, so they shouldn't fail the build as well. Once we get testing sorted out during the toolchain phase (or we restructure the build process in another way), then the circular dependencies issue may come back, but at that point we'll need to solve it globally for all packages anyway.